File: server-go/models/notification.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Notification struct {
	ID        bson.ObjectID  `bson:"_id,omitempty" json:"id"`
	User      bson.ObjectID  `bson:"user" json:"user"`
	FromUser  bson.ObjectID  `bson:"fromUser" json:"fromUser"`
	Type      string         `bson:"type" json:"type"`
	Post      *bson.ObjectID `bson:"post,omitempty" json:"post"`
	Read      bool           `bson:"read" json:"read"`
	CreatedAt time.Time      `bson:"createdAt" json:"createdAt"`
	UpdatedAt time.Time      `bson:"updatedAt" json:"updatedAt"`
}

File: server/models/Notification.js
const mongoose = require('mongoose');

const notificationSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  fromUser: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  type: { type: String, enum: ['like', 'retweet', 'follow', 'reply', 'mention'], required: true },
  post: { type: mongoose.Schema.Types.ObjectId, ref: 'Post', default: null },
  read: { type: Boolean, default: false },
}, { timestamps: true });

notificationSchema.index({ user: 1, createdAt: -1 });

module.exports = mongoose.model('Notification', notificationSchema);
